postgis/harden.sql

```diff
@@ -0,0 +1,64 @@
+-- This file replaces standard PostGIS functions with their versions
+-- actively checking input to be valid.
+
+begin;
+create or replace function _ST_FailOnInvalid(geom geometry)
+returns boolean as
+$$
+declare
+	v valid_detail;
+begin
+	v = ST_IsValidDetail(geom);
+	if (not v.valid) then
+		raise exception 'Input geometry of type % with % points is invalid at location %: %',
+		ST_GeometryType(geom), ST_NPoints(geom), ST_AsText(v.location), v.reason;
+	end if;
+	return true;
+end
+$$
+language plpgsql volatile strict parallel safe cost 1;
```

Comment:
I think this should be `IMMUTABLE`, not `VOLATILE`. For example, in a query like this:
```sql
Select * from my_table where ST_Intersection(geom, ST_MakeEnvelope(10, 10, 11, 11, 4326))
```
You want the envelope to be checked just once, not once per row.

The rest of the functions need also the rest of the properties set: `IMMUTABLE STRICT COST 2048`, and unless we drop support for releases prior to 9.6, the parallel property needs to be set in a configure step.

Reply:
It also seems to be not parallel safe, "cannot start commands in parallel scan" says postgres.